Improve comparison of intervals in `resolveFirstInterval()`

Ahmad-S792 opened the pull request:
#### 661903c7949859c5f2655b94ad9daf6d8d2404ec
<pre>
Improve comparison of intervals in `resolveFirstInterval()`
<a href="https://bugs.webkit.org/show_bug.cgi?id=250481">https://bugs.webkit.org/show_bug.cgi?id=250481</a>
<a href="https://rdar.apple.com/problem/104399900">rdar://problem/104399900</a>

Reviewed by Nikolas Zimmermann.

Inspired by: <a href="https://github.com/chromium/chromium/commit/5b93bdd34c17dae682ce14c45125abc603310b16">https://github.com/chromium/chromium/commit/5b93bdd34c17dae682ce14c45125abc603310b16</a>

For open-ended intervals such as [3) — i.e., begin = 3s and end/duration
unspecified — every call to resolveFirstInterval() would think a new interval
had been created and trigger unnecessary notifications and reschedules of
animations.

This patch compares the underlying SMILTime values directly using value().

The change preserves the original behavior but reduces redundant calls and
notifications. No new tests are added.

* Source/WebCore/svg/animation/SVGSMILElement.cpp:
(WebCore::SVGSMILElement::resolveFirstInterval):

Canonical link: <a href="https://commits.webkit.org/299110@main">https://commits.webkit.org/299110@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/763a3174cd2e7f57824fa70b301bb1aa02e2e6a9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/117861 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37539 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28171 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124005 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/69890 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/711becfc-2233-441c-9aad-8ed792be8ada) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/119739 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38231 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46121 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89439 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59059 "An unexpected error occured. Recent messages:Printed configuration") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/ccaf7aac-06d9-424a-9cd5-e1c2128385aa) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120813 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30440 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105677 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69934 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/8f9b8db4-a642-419f-8348-e8e2b7dc39af) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29504 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/23793 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67668 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99860 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23972 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127088 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44764 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33707 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98105 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45124 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/101902 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97893 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24909 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43286 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21257 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41184 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44636 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50310 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44095 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47441 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45785 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/763a3174cd2e7f57824fa70b301bb1aa02e2e6a9. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49818

Ahmad-S792 committed:
Improve comparison of intervals in `resolveFirstInterval()`
https://bugs.webkit.org/show_bug.cgi?id=250481
rdar://problem/104399900

Reviewed by Nikolas Zimmermann.

Inspired by: https://github.com/chromium/chromium/commit/5b93bdd34c17dae682ce14c45125abc603310b16

For open-ended intervals such as [3) — i.e., begin = 3s and end/duration
unspecified — every call to resolveFirstInterval() would think a new interval
had been created and trigger unnecessary notifications and reschedules of
animations.

This patch compares the underlying SMILTime values directly using value().

The change preserves the original behavior but reduces redundant calls and
notifications. No new tests are added.

* Source/WebCore/svg/animation/SVGSMILElement.cpp:
(WebCore::SVGSMILElement::resolveFirstInterval):

Canonical link: https://commits.webkit.org/299110@main
